test/e2e/framework/volume/fixtures.go

```diff
@@ -470,105 +468,128 @@ func TestVolumeClient(client clientset.Interface, config TestConfig, fsGroup *in
-			framework.ExpectNoError(err, "failed: getting the right fsType %s", fsType)
-		}
 	}
+	framework.ExpectNoError(e2epod.WaitForPodRunningInNamespace(client, clientPod))
+	testVolumeContent(client, clientPod, fsGroup, fsType, tests)
 }
 
-// InjectHTML inserts index.html with given content into given volume. It does so by
+// InjectContent inserts index.html with given content into given volume. It does so by
 // starting and auxiliary pod which writes the file there.
 // The volume must be writable.
-func InjectHTML(client clientset.Interface, config TestConfig, fsGroup *int64, volume v1.VolumeSource, content string) {
-	ginkgo.By(fmt.Sprint("starting ", config.Prefix, " injector"))
-	podClient := client.CoreV1().Pods(config.Namespace)
-	podName := fmt.Sprintf("%s-injector-%s", config.Prefix, rand.String(4))
-	volMountName := fmt.Sprintf("%s-volume-%s", config.Prefix, rand.String(4))
-	fileName := "/mnt/index.html"
-
-	injectPod := &v1.Pod{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "Pod",
-			APIVersion: "v1",
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name: podName,
-			Labels: map[string]string{
-				"role": config.Prefix + "-injector",
-			},
-		},
-		Spec: v1.PodSpec{
-			Containers: []v1.Container{
-				{
-					Name:    config.Prefix + "-injector",
-					Image:   GetTestImage(framework.BusyBoxImage),
-					Command: GenerateWriteFileCmd(content, fileName),
-					VolumeMounts: []v1.VolumeMount{
-						{
-							Name:      volMountName,
-							MountPath: "/mnt",
-						},
-					},
-					SecurityContext: GenerateSecurityContext(true),
-				},
-			},
-			SecurityContext: GeneratePodSecurityContext(fsGroup, nil),
-			RestartPolicy:   v1.RestartPolicyNever,
-			Volumes: []v1.Volume{
-				{
-					Name:         volMountName,
-					VolumeSource: volume,
-				},
-			},
-			NodeName:     config.ClientNodeName,
-			NodeSelector: config.NodeSelector,
-		},
+func InjectContent(client clientset.Interface, config TestConfig, fsGroup *int64, fsType string, tests []Test) {
+	injectorPod, err := runVolumeTesterPod(client, config, "injector", fsGroup, tests)
+	if err != nil {
+		e2elog.Failf("Failed to create injector pod: %v", err)
+		return
 	}
-
 	defer func() {
-		podClient.Delete(podName, nil)
+		e2epod.DeletePodOrFail(client, injectorPod.Namespace, injectorPod.Name)
+		e2epod.WaitForPodToDisappear(client, injectorPod.Namespace, injectorPod.Name, labels.Everything(), framework.Poll, framework.PodDeleteTimeout)
 	}()
 
-	injectPod, err := podClient.Create(injectPod)
-	framework.ExpectNoError(err, "Failed to create injector pod: %v", err)
-	err = e2epod.WaitForPodSuccessInNamespace(client, injectPod.Name, injectPod.Namespace)
-	framework.ExpectNoError(err)
+	ginkgo.By("Writing text file contents in the container.")
+	for i, test := range tests {
+		commands := []string{"exec", injectorPod.Name, fmt.Sprintf("--namespace=%v", injectorPod.Namespace), "--"}
+		if test.Mode == v1.PersistentVolumeBlock {
```

Comment:
Yes, I saw your `sha256sum` usage, but I decided not to use it as I would need to sum the data in the injector pod and parse it from pod output to be able to check it in the reader pod later. `echo` works well here (not sure how well it will work on Windows).